File: Algoritmos de Ordenación en Python/Intercambio.py
#UNIVERSIDAD DE LAS FUERZAS ARMADAS- ESPE
#Autor:Pamela Jesabel Carriel Mier 
#Ordenacion por intercambio en python
#Fecha : 05 de junio del 2023
def bubble_sort(lista):
    n = len(lista)
    #Se utiliza el bucle for para iterar sobre la lista, y dentro de este bucle
    for i in range(n - 1):
        #e utiliza una variable intercambio para verificar si se ha realizado algún intercambio en cada iteración del bucle interno.
        intercambio = False
        #para comparar y realizar intercambios entre elementos adyacentes. 
        for j in range(n - 1 - i):
            if lista[j] > lista[j + 1]:
                lista[j], lista[j + 1] = lista[j + 1], lista[j]
                intercambio = True
        if not intercambio:
            break
    return lista

lista = [5, 2, 7, 10, 1, 0]
print("Lista original:", lista)
lista_ordenada = bubble_sort(lista)
print("Lista ordenada:", lista_ordenada)

File: Algoritmos de Ordenación en JAVA/Intercambio.java
/*UNIVERSIDAD DE LAS FUERZAS ARMADAS- ESPE
Autor:Pamela Jesabel Carriel Mier 
Ordenacion por intercambio 
Fecha : 05 de junio del 2023*/
public class Intercambio{
    public static void bubbleSort(int[] array) {
        int n = array.length;
        boolean intercambio;
        
        for (int i = 0; i < n - 1; i++) {
            intercambio = false;
            
            for (int j = 0; j < n - 1 - i; j++) {
                if (array[j] > array[j + 1]) {
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                    intercambio = true;
                }
            }
            
            if (!intercambio) {
                break;
            }
        }
    }
    
    // Ejemplo de uso
    public static void main(String[] args) {
        int[] array = {5, 2, 8, 12, 1, 6};
        
        System.out.println("Lista original:");
        imprimirArray(array);
        
        bubbleSort(array);
        
        System.out.println("Lista ordenada:");
        imprimirArray(array);
    }
    
    public static void imprimirArray(int[] array) {
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + " ");
        }
        System.out.println();
    }
}
